### PR TITLE
[FIX] l10n_it_edi_sdicoop: fix access rights so demo user can send in…

### PR DESCRIPTION
**Pull request by jco-odoo**
…voices

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class AccountEdiFormat(models.Model):
     def _cron_receive_fattura_pa(self):
         ''' Check the proxy for incoming invoices.
         '''
-        proxy_users = self.env['account_edi_proxy_client.user'].search([('edi_format_id', '=', self.env.ref('l10n_it_edi.edi_fatturaPA').id)])
+        proxy_users = self.env['account_edi_proxy_client.user'].sudo().search([('edi_format_id', '=', self.env.ref('l10n_it_edi.edi_fatturaPA').id)])
```
Comment (william-andre) [inline addons/l10n_it_edi_sdicoop/models/account_edi_format.py:25]: Please try to reduce the scope of the sudo environment

**Review by william-andre — CHANGES_REQUESTED** [review]
I suppose that there is sensitive data on that model that regular users shouldn't be able to read?

If not, update the access rights instead.

Also, the commit message is pretty bad; you don't want to fix it for the demo, but for any user that is not database administrator...

File: addons/account_edi_proxy_client/models/account_edi_format.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ def _get_proxy_user(self, company):
         '''Returns the proxy_user associated with this edi format.
         '''
         self.ensure_one()
-        return company.account_edi_proxy_client_ids.filtered(lambda u: u.edi_format_id == self)
+        return company.sudo().account_edi_proxy_client_ids.filtered(lambda u: u.edi_format_id == self)
```
[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/account_edi_proxy_client/models/account_edi_format.py:16]: Maybe, but the easiest way to solve it as the proxy client user is only accessible by base.group_system.

[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/account_edi_proxy_client/models/account_edi_format.py:16]: Anyways, if the user gets an access error, the cron can still take it.

[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account_edi_proxy_client/models/account_edi_format.py:16]: > Maybe, but the easiest way to solve it as the proxy client user is only accessible by base.group_system.

Not sure I understand what you mean

[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/account_edi_proxy_client/models/account_edi_format.py:16]: That if we do not sudo the record, a regular user can not read it.

[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account_edi_proxy_client/models/account_edi_format.py:16]: That I know... It doesn't mean that you should just sudo blindly any record causing an issue like that

**Comment by jco-odoo** [comment]
@william-andre Still WIP, but I think this principle is better where we only need to ask explicitly when the invoicing user wants to read the private fields or write to them.

**Comment by william-andre** [comment]
It looks a lot better indeed

**Comment by jco-odoo** [comment]
@william-andre tested by @dbkosky and me and I checked again the cases where it needs to write or check refresh_token/private_key, so this should be good to merge.

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**